orcid-message-listener/src/main/java/org/orcid/listener/solr/SolrMessageProcessor.java

```diff
@@ -83,6 +83,12 @@ private void updateSolrIndex(String orcid) {
         }
         try{
             org.orcid.jaxb.model.record_v2.Record record = orcid20ApiClient.fetchPublicRecord(orcid); 
+            
+            // Remove deactivated records from SOLR index
+            if (record.getHistory() != null && record.getHistory().getDeactivationDate() != null && record.getHistory().getDeactivationDate().getValue() != null) {
+                solrUpdater.deleteById(orcid);
```

Comment:
You check for a deactivated exception below.  Are both checks necessary?

Reply:
Na @TomDemeranville, deactivated records don't throw exceptions, they just return an empty Record with the deactivation date, that's why we should look for it manually.

The exceptions below catch locked or deprecated records.

Reply:
My mistake!